### PR TITLE
Implement Socket.socketpair to return Socket instances

### PR DESCRIPTION
**Pull request by etehtsea**

### CODE REVIEW

File: spec/ruby/library/socket/shared/socketpair.rb [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
       s1.close
       s2.close
     end
+
+    it "responses with array of two sockets" do
+      s1, s2 = Socket.public_send(@method, :UNIX, :STREAM)
+
+      s1.should be_an_instance_of(Socket)
+      s2.should be_an_instance_of(Socket)
+    end
```
Comment (headius) [inline spec/ruby/library/socket/shared/socketpair.rb]: @etehtsea We usually update from ruby/spec via @eregon's process of migrating subtree commits back and forth. I'm not sure, but this separate commit may interfere with that process.

File: spec/ruby/library/socket/shared/socketpair.rb [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
       s2.gets.should == "test\n"
       s1.close
       s2.close
     end
+
+    it "responses with array of two sockets" do
+      s1, s2 = Socket.public_send(@method, :UNIX, :STREAM)
+
+      s1.should be_an_instance_of(Socket)
+      s2.should be_an_instance_of(Socket)
```
Comment (eregon) [inline spec/ruby/library/socket/shared/socketpair.rb]: The opened file descriptors should be closed here.

Comment (etehtsea) [inline spec/ruby/library/socket/shared/socketpair.rb]: done

**Comment by etehtsea** [comment]
Done.

**Pull request merged**